# The Adaptation Project - First Generation

## Abstract
Microbes adapt as they establish within a host microbiome, yet this crucial interaction, also essential for the development of live bacterial therapeutics, is under-explored. Long-term in vitro bacterial evolution and adaptation studies of Escherichia coli (E. coli) found many genome modifications, even in minimal media. However, in vivo studies face challenges, including most probiotic strains failing to persist in the gut and concerns about oversimplification that germ-free or antibiotic-depleted models offer compared to normal mammalian microbiomes. Here we overcome these challenges by using host-adapted probiotic strains to assess long-term colonization in an animal with an unaltered microbiome and we show the in vivo adaptation of over 3100 bacterial isolates over 10 months (approximately 5,200 generations) for mouse-native (EcAZ) and non-native (Nissle1917, human derived) E. coli strains. The non-native strain was less successful at long-term engraftment persistence (38% of non-native mice lost colonization over time, compared to 0% for the native strain), had a less stable core genome (4.6x more pangenome variance; approximately twice as many unique SNPs), increased plasmid copy number variation, and underwent more phage predation than the native strain. Our study establishes that non-host-native probiotics can exhibit significant genomic instability during long-term colonization, which may have implications on their potential use as long-term therapeutics.

## Organization

Directories contain relevant scripts for the cluster, output files, and notebooks with python code for figure generation. Not all graphs were used in the paper. Directories organized by topic.


## Relevant Links

### BioRxiv
doi:

### Final Journal
doi:

### Original Data Availability
EBI accession: ERP145794
Qiita ID: 12986
