{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "df85fa10",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9a29b8eb",
   "metadata": {},
   "source": [
    "# AZ20"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "94460aad",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Marker lineage</th>\n",
       "      <th># genomes</th>\n",
       "      <th># markers</th>\n",
       "      <th># marker sets</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5+</th>\n",
       "      <th>...</th>\n",
       "      <th>LA75</th>\n",
       "      <th>LGA75</th>\n",
       "      <th>ID</th>\n",
       "      <th>status</th>\n",
       "      <th>superkingdom</th>\n",
       "      <th>phylum</th>\n",
       "      <th>class</th>\n",
       "      <th>order</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>23B_Day7_11_S38</th>\n",
       "      <td>Bacteria</td>\n",
       "      <td>5449</td>\n",
       "      <td>104</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "      <td>104</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>AZ20_assemblies/23B_Day7_11_S38.fasta</td>\n",
       "      <td>found</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales</td>\n",
       "      <td>Enterobacteriaceae</td>\n",
       "      <td>Escherichia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23B_Day7_12_S384</th>\n",
       "      <td>Bacteria</td>\n",
       "      <td>5449</td>\n",
       "      <td>104</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "      <td>104</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>AZ20_assemblies/23B_Day7_12_S384.fasta</td>\n",
       "      <td>found</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales</td>\n",
       "      <td>Enterobacteriaceae</td>\n",
       "      <td>Escherichia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23B_Day7_13_S1294</th>\n",
       "      <td>Bacteria</td>\n",
       "      <td>5449</td>\n",
       "      <td>104</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "      <td>86</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>AZ20_assemblies/23B_Day7_13_S1294.fasta</td>\n",
       "      <td>found</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales</td>\n",
       "      <td>Enterobacteriaceae</td>\n",
       "      <td>Escherichia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23B_Day7_14_S1295</th>\n",
       "      <td>Bacteria</td>\n",
       "      <td>5449</td>\n",
       "      <td>104</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "      <td>101</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>AZ20_assemblies/23B_Day7_14_S1295.fasta</td>\n",
       "      <td>found</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales</td>\n",
       "      <td>Enterobacteriaceae</td>\n",
       "      <td>Escherichia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23B_Day7_16_S1297</th>\n",
       "      <td>Bacteria</td>\n",
       "      <td>5449</td>\n",
       "      <td>104</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "      <td>95</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>AZ20_assemblies/23B_Day7_16_S1297.fasta</td>\n",
       "      <td>found</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales</td>\n",
       "      <td>Enterobacteriaceae</td>\n",
       "      <td>Escherichia</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 75 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Marker lineage  # genomes  # markers  # marker sets  0    1  \\\n",
       "23B_Day7_11_S38         Bacteria       5449        104             58  0  104   \n",
       "23B_Day7_12_S384        Bacteria       5449        104             58  0  104   \n",
       "23B_Day7_13_S1294       Bacteria       5449        104             58  0   86   \n",
       "23B_Day7_14_S1295       Bacteria       5449        104             58  0  101   \n",
       "23B_Day7_16_S1297       Bacteria       5449        104             58  0   95   \n",
       "\n",
       "                    2  3  4  5+  ...  LA75  LGA75  \\\n",
       "23B_Day7_11_S38     0  0  0   0  ...    11     11   \n",
       "23B_Day7_12_S384    0  0  0   0  ...     1      1   \n",
       "23B_Day7_13_S1294  18  0  0   0  ...     1      1   \n",
       "23B_Day7_14_S1295   3  0  0   0  ...     2      2   \n",
       "23B_Day7_16_S1297   9  0  0   0  ...     1      1   \n",
       "\n",
       "                                                        ID  status  \\\n",
       "23B_Day7_11_S38      AZ20_assemblies/23B_Day7_11_S38.fasta   found   \n",
       "23B_Day7_12_S384    AZ20_assemblies/23B_Day7_12_S384.fasta   found   \n",
       "23B_Day7_13_S1294  AZ20_assemblies/23B_Day7_13_S1294.fasta   found   \n",
       "23B_Day7_14_S1295  AZ20_assemblies/23B_Day7_14_S1295.fasta   found   \n",
       "23B_Day7_16_S1297  AZ20_assemblies/23B_Day7_16_S1297.fasta   found   \n",
       "\n",
       "                   superkingdom          phylum                class  \\\n",
       "23B_Day7_11_S38        Bacteria  Proteobacteria  Gammaproteobacteria   \n",
       "23B_Day7_12_S384       Bacteria  Proteobacteria  Gammaproteobacteria   \n",
       "23B_Day7_13_S1294      Bacteria  Proteobacteria  Gammaproteobacteria   \n",
       "23B_Day7_14_S1295      Bacteria  Proteobacteria  Gammaproteobacteria   \n",
       "23B_Day7_16_S1297      Bacteria  Proteobacteria  Gammaproteobacteria   \n",
       "\n",
       "                              order              family        genus  \n",
       "23B_Day7_11_S38    Enterobacterales  Enterobacteriaceae  Escherichia  \n",
       "23B_Day7_12_S384   Enterobacterales  Enterobacteriaceae  Escherichia  \n",
       "23B_Day7_13_S1294  Enterobacterales  Enterobacteriaceae  Escherichia  \n",
       "23B_Day7_14_S1295  Enterobacterales  Enterobacteriaceae  Escherichia  \n",
       "23B_Day7_16_S1297  Enterobacterales  Enterobacteriaceae  Escherichia  \n",
       "\n",
       "[5 rows x 75 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Nissle_allqc = pd.read_csv('qc_output_files/AZ20_qc_all.csv', index_col=0)\n",
    "Nissle_allqc.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "42f3c9f7",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/m1/h11rhh850_s7fjxgxw5wzh_r0000gp/T/ipykernel_51169/218532717.py:2: FutureWarning: In a future version of pandas all arguments of StringMethods.rsplit except for the argument 'pat' will be keyword-only.\n",
      "  Nissle_allqc2['sample-id'] = Nissle_allqc2['index'].str.rsplit('_', 1).str[0]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>Marker lineage</th>\n",
       "      <th># genomes</th>\n",
       "      <th># markers</th>\n",
       "      <th># marker sets</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>...</th>\n",
       "      <th>LGA75</th>\n",
       "      <th>ID</th>\n",
       "      <th>status</th>\n",
       "      <th>superkingdom</th>\n",
       "      <th>phylum</th>\n",
       "      <th>class</th>\n",
       "      <th>order</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "      <th>sample-id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>23B_Day7_11_S38</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>5449</td>\n",
       "      <td>104</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "      <td>104</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>11</td>\n",
       "      <td>AZ20_assemblies/23B_Day7_11_S38.fasta</td>\n",
       "      <td>found</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales</td>\n",
       "      <td>Enterobacteriaceae</td>\n",
       "      <td>Escherichia</td>\n",
       "      <td>23B_Day7_11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>23B_Day7_12_S384</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>5449</td>\n",
       "      <td>104</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "      <td>104</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>AZ20_assemblies/23B_Day7_12_S384.fasta</td>\n",
       "      <td>found</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales</td>\n",
       "      <td>Enterobacteriaceae</td>\n",
       "      <td>Escherichia</td>\n",
       "      <td>23B_Day7_12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>23B_Day7_13_S1294</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>5449</td>\n",
       "      <td>104</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "      <td>86</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>AZ20_assemblies/23B_Day7_13_S1294.fasta</td>\n",
       "      <td>found</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales</td>\n",
       "      <td>Enterobacteriaceae</td>\n",
       "      <td>Escherichia</td>\n",
       "      <td>23B_Day7_13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>23B_Day7_14_S1295</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>5449</td>\n",
       "      <td>104</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "      <td>101</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>AZ20_assemblies/23B_Day7_14_S1295.fasta</td>\n",
       "      <td>found</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales</td>\n",
       "      <td>Enterobacteriaceae</td>\n",
       "      <td>Escherichia</td>\n",
       "      <td>23B_Day7_14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>23B_Day7_16_S1297</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>5449</td>\n",
       "      <td>104</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "      <td>95</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>AZ20_assemblies/23B_Day7_16_S1297.fasta</td>\n",
       "      <td>found</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales</td>\n",
       "      <td>Enterobacteriaceae</td>\n",
       "      <td>Escherichia</td>\n",
       "      <td>23B_Day7_16</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 77 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               index Marker lineage  # genomes  # markers  # marker sets  0  \\\n",
       "0    23B_Day7_11_S38       Bacteria       5449        104             58  0   \n",
       "1   23B_Day7_12_S384       Bacteria       5449        104             58  0   \n",
       "2  23B_Day7_13_S1294       Bacteria       5449        104             58  0   \n",
       "3  23B_Day7_14_S1295       Bacteria       5449        104             58  0   \n",
       "4  23B_Day7_16_S1297       Bacteria       5449        104             58  0   \n",
       "\n",
       "     1   2  3  4  ...  LGA75                                       ID  status  \\\n",
       "0  104   0  0  0  ...     11    AZ20_assemblies/23B_Day7_11_S38.fasta   found   \n",
       "1  104   0  0  0  ...      1   AZ20_assemblies/23B_Day7_12_S384.fasta   found   \n",
       "2   86  18  0  0  ...      1  AZ20_assemblies/23B_Day7_13_S1294.fasta   found   \n",
       "3  101   3  0  0  ...      2  AZ20_assemblies/23B_Day7_14_S1295.fasta   found   \n",
       "4   95   9  0  0  ...      1  AZ20_assemblies/23B_Day7_16_S1297.fasta   found   \n",
       "\n",
       "   superkingdom          phylum                class             order  \\\n",
       "0      Bacteria  Proteobacteria  Gammaproteobacteria  Enterobacterales   \n",
       "1      Bacteria  Proteobacteria  Gammaproteobacteria  Enterobacterales   \n",
       "2      Bacteria  Proteobacteria  Gammaproteobacteria  Enterobacterales   \n",
       "3      Bacteria  Proteobacteria  Gammaproteobacteria  Enterobacterales   \n",
       "4      Bacteria  Proteobacteria  Gammaproteobacteria  Enterobacterales   \n",
       "\n",
       "               family        genus    sample-id  \n",
       "0  Enterobacteriaceae  Escherichia  23B_Day7_11  \n",
       "1  Enterobacteriaceae  Escherichia  23B_Day7_12  \n",
       "2  Enterobacteriaceae  Escherichia  23B_Day7_13  \n",
       "3  Enterobacteriaceae  Escherichia  23B_Day7_14  \n",
       "4  Enterobacteriaceae  Escherichia  23B_Day7_16  \n",
       "\n",
       "[5 rows x 77 columns]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Nissle_allqc2 = Nissle_allqc.reset_index()\n",
    "Nissle_allqc2['sample-id'] = Nissle_allqc2['index'].str.rsplit('_', 1).str[0]\n",
    "Nissle_allqc2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "33c3af10",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['index', 'Marker lineage', '# genomes', '# markers', '# marker sets',\n",
       "       '0', '1', '2', '3', '4', '5+', 'Completeness', 'Contamination',\n",
       "       'Strain heterogeneity', '# unique markers (of 43)', '# multi-copy',\n",
       "       'Taxonomy', '# contigs (>= 0 bp)', '# contigs (>= 1000 bp)',\n",
       "       '# contigs (>= 5000 bp)', '# contigs (>= 10000 bp)',\n",
       "       '# contigs (>= 25000 bp)', '# contigs (>= 50000 bp)',\n",
       "       'Total length (>= 0 bp)', 'Total length (>= 1000 bp)',\n",
       "       'Total length (>= 5000 bp)', 'Total length (>= 10000 bp)',\n",
       "       'Total length (>= 25000 bp)', 'Total length (>= 50000 bp)', '# contigs',\n",
       "       'Largest contig', 'Total length', 'Reference length', 'GC (%)',\n",
       "       'Reference GC (%)', 'N50', 'NG50', 'N75', 'NG75', 'L50', 'LG50', 'L75',\n",
       "       'LG75', '# misassemblies', '# misassembled contigs',\n",
       "       'Misassembled contigs length', '# local misassemblies',\n",
       "       '# scaffold gap ext. mis.', '# scaffold gap loc. mis.',\n",
       "       '# unaligned mis. contigs', '# unaligned contigs', 'Unaligned length',\n",
       "       'Genome fraction (%)', 'Duplication ratio', '# N's per 100 kbp',\n",
       "       '# mismatches per 100 kbp', '# indels per 100 kbp',\n",
       "       '# genomic features', 'Largest alignment', 'Total aligned length',\n",
       "       'NA50', 'NGA50', 'NA75', 'NGA75', 'LA50', 'LGA50', 'LA75', 'LGA75',\n",
       "       'ID', 'status', 'superkingdom', 'phylum', 'class', 'order', 'family',\n",
       "       'genus', 'sample-id'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Nissle_allqc2.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "id": "b6d81ca7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample-id</th>\n",
       "      <th>Completeness</th>\n",
       "      <th>Contamination</th>\n",
       "      <th># contigs</th>\n",
       "      <th># contigs (&gt;= 50000 bp)</th>\n",
       "      <th>Largest contig</th>\n",
       "      <th>Total length</th>\n",
       "      <th>Reference length</th>\n",
       "      <th>Total aligned length</th>\n",
       "      <th>Genome fraction (%)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>23B_Day7_11</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>89</td>\n",
       "      <td>17</td>\n",
       "      <td>670168</td>\n",
       "      <td>4982223</td>\n",
       "      <td>5064668</td>\n",
       "      <td>4982220</td>\n",
       "      <td>98.370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>23B_Day7_12</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>5056481</td>\n",
       "      <td>5065168</td>\n",
       "      <td>5064668</td>\n",
       "      <td>5063920</td>\n",
       "      <td>99.921</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>23B_Day7_13</td>\n",
       "      <td>100.0</td>\n",
       "      <td>13.79</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>5054136</td>\n",
       "      <td>5475531</td>\n",
       "      <td>5064668</td>\n",
       "      <td>5471892</td>\n",
       "      <td>99.869</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>23B_Day7_14</td>\n",
       "      <td>100.0</td>\n",
       "      <td>5.17</td>\n",
       "      <td>16</td>\n",
       "      <td>8</td>\n",
       "      <td>2787855</td>\n",
       "      <td>5363807</td>\n",
       "      <td>5064668</td>\n",
       "      <td>5360446</td>\n",
       "      <td>99.880</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>23B_Day7_16</td>\n",
       "      <td>100.0</td>\n",
       "      <td>12.07</td>\n",
       "      <td>17</td>\n",
       "      <td>4</td>\n",
       "      <td>5046245</td>\n",
       "      <td>5554755</td>\n",
       "      <td>5064668</td>\n",
       "      <td>5552302</td>\n",
       "      <td>99.776</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     sample-id  Completeness  Contamination  # contigs  \\\n",
       "0  23B_Day7_11         100.0           0.00         89   \n",
       "1  23B_Day7_12         100.0           0.00          3   \n",
       "2  23B_Day7_13         100.0          13.79         31   \n",
       "3  23B_Day7_14         100.0           5.17         16   \n",
       "4  23B_Day7_16         100.0          12.07         17   \n",
       "\n",
       "   # contigs (>= 50000 bp)  Largest contig  Total length  Reference length  \\\n",
       "0                       17          670168       4982223           5064668   \n",
       "1                        1         5056481       5065168           5064668   \n",
       "2                        1         5054136       5475531           5064668   \n",
       "3                        8         2787855       5363807           5064668   \n",
       "4                        4         5046245       5554755           5064668   \n",
       "\n",
       "   Total aligned length  Genome fraction (%)  \n",
       "0               4982220               98.370  \n",
       "1               5063920               99.921  \n",
       "2               5471892               99.869  \n",
       "3               5360446               99.880  \n",
       "4               5552302               99.776  "
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Nissle_primqc = Nissle_allqc2[['sample-id', 'Completeness', 'Contamination', \n",
    "                              '# contigs', '# contigs (>= 50000 bp)', 'Largest contig', \n",
    "                              'Total length', 'Reference length', 'Total aligned length', \n",
    "                              'Genome fraction (%)']].copy()\n",
    "Nissle_primqc.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "ebcfe8b8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1222, 10)"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Nissle_primqc.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "7e8c37a6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>23B_Day7_10_S382/large_plasmids.fasta:3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>23B_Day7_11_S383/large_plasmids.fasta:3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>23B_Day7_12_S384/large_plasmids.fasta:3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>23B_Day7_13_S1294/large_plasmids.fasta:2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>23B_Day7_14_S1295/large_plasmids.fasta:2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1470</th>\n",
       "      <td>30N_Day7_6_S924/large_plasmids.fasta:2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1471</th>\n",
       "      <td>30N_Day7_7_S925/large_plasmids.fasta:2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1472</th>\n",
       "      <td>30N_Day7_8_S926/large_plasmids.fasta:2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1473</th>\n",
       "      <td>30N_Day7_9_S927/large_plasmids.fasta:2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1474</th>\n",
       "      <td>AZ20_ref/large_plasmids.fasta:2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1475 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             0\n",
       "0      23B_Day7_10_S382/large_plasmids.fasta:3\n",
       "1      23B_Day7_11_S383/large_plasmids.fasta:3\n",
       "2      23B_Day7_12_S384/large_plasmids.fasta:3\n",
       "3     23B_Day7_13_S1294/large_plasmids.fasta:2\n",
       "4     23B_Day7_14_S1295/large_plasmids.fasta:2\n",
       "...                                        ...\n",
       "1470    30N_Day7_6_S924/large_plasmids.fasta:2\n",
       "1471    30N_Day7_7_S925/large_plasmids.fasta:2\n",
       "1472    30N_Day7_8_S926/large_plasmids.fasta:2\n",
       "1473    30N_Day7_9_S927/large_plasmids.fasta:2\n",
       "1474           AZ20_ref/large_plasmids.fasta:2\n",
       "\n",
       "[1475 rows x 1 columns]"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Nissle_plasmid_counts = pd.read_csv('../plasmids/AZ20_lplasmid_number.txt', sep='\\t', header=None)\n",
    "Nissle_plasmid_counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "4752237d",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/m1/h11rhh850_s7fjxgxw5wzh_r0000gp/T/ipykernel_51169/2585858066.py:1: FutureWarning: In a future version of pandas all arguments of StringMethods.split except for the argument 'pat' will be keyword-only.\n",
      "  Nissle_plasmid_counts[['sample', 'other']] = Nissle_plasmid_counts[0].str.split('/', 1, expand=True)\n"
     ]
    }
   ],
   "source": [
    "Nissle_plasmid_counts[['sample', 'other']] = Nissle_plasmid_counts[0].str.split('/', 1, expand=True)\n",
    "Nissle_plasmid_counts['number_of_plasmids'] = Nissle_plasmid_counts['other'].str.split(':').str[1]\n",
    "Nissle_plasmid_counts['sample-id'] = Nissle_plasmid_counts['sample'].replace(to_replace='_S[0-9]*', value='', regex=True)\n",
    "Nissle_plasmid_counts['sample-id'] = Nissle_plasmid_counts['sample-id'].replace(to_replace='_L003', value='', regex=True)\n",
    "Nissle_plasmid_counts['sample-id'] = Nissle_plasmid_counts['sample-id'].replace(to_replace='_pos', value='p', regex=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "id": "7c8585bc",
   "metadata": {},
   "outputs": [],
   "source": [
    "for x in range(1,13):\n",
    "    Nissle_plasmid_counts['sample-id'] = Nissle_plasmid_counts['sample-id'].replace(to_replace='_%s_2' % x, value='_%sx' % x, regex=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "dcbd1aba",
   "metadata": {},
   "outputs": [],
   "source": [
    "Nissle_plasmid_counts['sample-id'] = Nissle_plasmid_counts['sample-id'].replace(to_replace='_12_A', value='_12', regex=True)\n",
    "Nissle_plasmid_counts['sample-id'] = Nissle_plasmid_counts['sample-id'].replace(to_replace='_12_B', value='_13', regex=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "id": "25a763cb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample-id</th>\n",
       "      <th>number_of_plasmids</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>23B_Day7_10</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>23B_Day7_11</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>23B_Day7_12</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>23B_Day7_13</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>23B_Day7_14</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     sample-id number_of_plasmids\n",
       "0  23B_Day7_10                  3\n",
       "1  23B_Day7_11                  3\n",
       "2  23B_Day7_12                  3\n",
       "3  23B_Day7_13                  2\n",
       "4  23B_Day7_14                  2"
      ]
     },
     "execution_count": 108,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Nissle_plasmid_counts2 = Nissle_plasmid_counts[['sample-id', 'number_of_plasmids']].copy()\n",
    "Nissle_plasmid_counts2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "de075079",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample-id</th>\n",
       "      <th>Completeness</th>\n",
       "      <th>Contamination</th>\n",
       "      <th># contigs</th>\n",
       "      <th># contigs (&gt;= 50000 bp)</th>\n",
       "      <th>Largest contig</th>\n",
       "      <th>Total length</th>\n",
       "      <th>Reference length</th>\n",
       "      <th>Total aligned length</th>\n",
       "      <th>Genome fraction (%)</th>\n",
       "      <th>number_of_plasmids</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>23B_Day7_11</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>89.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>670168.0</td>\n",
       "      <td>4982223.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>4982220.0</td>\n",
       "      <td>98.370</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>23B_Day7_12</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5056481.0</td>\n",
       "      <td>5065168.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>5063920.0</td>\n",
       "      <td>99.921</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>23B_Day7_13</td>\n",
       "      <td>100.0</td>\n",
       "      <td>13.79</td>\n",
       "      <td>31.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5054136.0</td>\n",
       "      <td>5475531.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>5471892.0</td>\n",
       "      <td>99.869</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>23B_Day7_14</td>\n",
       "      <td>100.0</td>\n",
       "      <td>5.17</td>\n",
       "      <td>16.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>2787855.0</td>\n",
       "      <td>5363807.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>5360446.0</td>\n",
       "      <td>99.880</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>23B_Day7_16</td>\n",
       "      <td>100.0</td>\n",
       "      <td>12.07</td>\n",
       "      <td>17.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5046245.0</td>\n",
       "      <td>5554755.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>5552302.0</td>\n",
       "      <td>99.776</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1535</th>\n",
       "      <td>29R_Month3_12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1536</th>\n",
       "      <td>29R_Month3_17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1537</th>\n",
       "      <td>29R_Month3_19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1538</th>\n",
       "      <td>30N_Day7_10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1539</th>\n",
       "      <td>AZ20_ref</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1540 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          sample-id  Completeness  Contamination  # contigs  \\\n",
       "0       23B_Day7_11         100.0           0.00       89.0   \n",
       "1       23B_Day7_12         100.0           0.00        3.0   \n",
       "2       23B_Day7_13         100.0          13.79       31.0   \n",
       "3       23B_Day7_14         100.0           5.17       16.0   \n",
       "4       23B_Day7_16         100.0          12.07       17.0   \n",
       "...             ...           ...            ...        ...   \n",
       "1535  29R_Month3_12           NaN            NaN        NaN   \n",
       "1536  29R_Month3_17           NaN            NaN        NaN   \n",
       "1537  29R_Month3_19           NaN            NaN        NaN   \n",
       "1538    30N_Day7_10           NaN            NaN        NaN   \n",
       "1539       AZ20_ref           NaN            NaN        NaN   \n",
       "\n",
       "      # contigs (>= 50000 bp)  Largest contig  Total length  Reference length  \\\n",
       "0                        17.0        670168.0     4982223.0         5064668.0   \n",
       "1                         1.0       5056481.0     5065168.0         5064668.0   \n",
       "2                         1.0       5054136.0     5475531.0         5064668.0   \n",
       "3                         8.0       2787855.0     5363807.0         5064668.0   \n",
       "4                         4.0       5046245.0     5554755.0         5064668.0   \n",
       "...                       ...             ...           ...               ...   \n",
       "1535                      NaN             NaN           NaN               NaN   \n",
       "1536                      NaN             NaN           NaN               NaN   \n",
       "1537                      NaN             NaN           NaN               NaN   \n",
       "1538                      NaN             NaN           NaN               NaN   \n",
       "1539                      NaN             NaN           NaN               NaN   \n",
       "\n",
       "      Total aligned length  Genome fraction (%) number_of_plasmids  \n",
       "0                4982220.0               98.370                  3  \n",
       "1                5063920.0               99.921                  3  \n",
       "2                5471892.0               99.869                  2  \n",
       "3                5360446.0               99.880                  2  \n",
       "4                5552302.0               99.776                  2  \n",
       "...                    ...                  ...                ...  \n",
       "1535                   NaN                  NaN                  3  \n",
       "1536                   NaN                  NaN                  3  \n",
       "1537                   NaN                  NaN                  3  \n",
       "1538                   NaN                  NaN                  2  \n",
       "1539                   NaN                  NaN                  2  \n",
       "\n",
       "[1540 rows x 11 columns]"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Nissle_qc_plas = pd.merge(Nissle_primqc, Nissle_plasmid_counts2, how='outer', \n",
    "                          left_on='sample-id', right_on='sample-id')\n",
    "Nissle_qc_plas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "id": "ea25085c",
   "metadata": {},
   "outputs": [],
   "source": [
    "Nissle_removal = pd.read_csv('../AZ20_bad_genomes.txt')\n",
    "Nissle_bad = Nissle_removal.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "id": "b2ab3df1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1508, 12)"
      ]
     },
     "execution_count": 163,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Nissle_qc_plas = Nissle_qc_plas[~(Nissle_qc_plas['sample-id'].isin(Nissle_bad))].copy()\n",
    "Nissle_qc_plas.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "id": "5c5a6c44",
   "metadata": {},
   "outputs": [],
   "source": [
    "Nissle_qc_plas['number_of_plasmids'] = Nissle_qc_plas['number_of_plasmids'].fillna(0)\n",
    "#makes sure to account for everything that exists, even when it might not have detected plasmids\n",
    "Nissle_qc_plas['number_of_plasmids'] = Nissle_qc_plas['number_of_plasmids'].astype(int)\n",
    "#values treated as numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "id": "98354d7d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample-id</th>\n",
       "      <th>Completeness</th>\n",
       "      <th>Contamination</th>\n",
       "      <th># contigs</th>\n",
       "      <th># contigs (&gt;= 50000 bp)</th>\n",
       "      <th>Largest contig</th>\n",
       "      <th>Total length</th>\n",
       "      <th>Reference length</th>\n",
       "      <th>Total aligned length</th>\n",
       "      <th>Genome fraction (%)</th>\n",
       "      <th>number_of_plasmids</th>\n",
       "      <th>ideal_contig_num</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>23B_Day7_11</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>89.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>670168.0</td>\n",
       "      <td>4982223.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>4982220.0</td>\n",
       "      <td>98.370</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>23B_Day7_12</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5056481.0</td>\n",
       "      <td>5065168.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>5063920.0</td>\n",
       "      <td>99.921</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>23B_Day7_13</td>\n",
       "      <td>100.0</td>\n",
       "      <td>13.79</td>\n",
       "      <td>31.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5054136.0</td>\n",
       "      <td>5475531.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>5471892.0</td>\n",
       "      <td>99.869</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>23B_Day7_14</td>\n",
       "      <td>100.0</td>\n",
       "      <td>5.17</td>\n",
       "      <td>16.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>2787855.0</td>\n",
       "      <td>5363807.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>5360446.0</td>\n",
       "      <td>99.880</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>23B_Day7_16</td>\n",
       "      <td>100.0</td>\n",
       "      <td>12.07</td>\n",
       "      <td>17.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5046245.0</td>\n",
       "      <td>5554755.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>5552302.0</td>\n",
       "      <td>99.776</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     sample-id  Completeness  Contamination  # contigs  \\\n",
       "0  23B_Day7_11         100.0           0.00       89.0   \n",
       "1  23B_Day7_12         100.0           0.00        3.0   \n",
       "2  23B_Day7_13         100.0          13.79       31.0   \n",
       "3  23B_Day7_14         100.0           5.17       16.0   \n",
       "4  23B_Day7_16         100.0          12.07       17.0   \n",
       "\n",
       "   # contigs (>= 50000 bp)  Largest contig  Total length  Reference length  \\\n",
       "0                     17.0        670168.0     4982223.0         5064668.0   \n",
       "1                      1.0       5056481.0     5065168.0         5064668.0   \n",
       "2                      1.0       5054136.0     5475531.0         5064668.0   \n",
       "3                      8.0       2787855.0     5363807.0         5064668.0   \n",
       "4                      4.0       5046245.0     5554755.0         5064668.0   \n",
       "\n",
       "   Total aligned length  Genome fraction (%)  number_of_plasmids  \\\n",
       "0             4982220.0               98.370                   3   \n",
       "1             5063920.0               99.921                   3   \n",
       "2             5471892.0               99.869                   2   \n",
       "3             5360446.0               99.880                   2   \n",
       "4             5552302.0               99.776                   2   \n",
       "\n",
       "   ideal_contig_num  \n",
       "0                 4  \n",
       "1                 4  \n",
       "2                 3  \n",
       "3                 3  \n",
       "4                 3  "
      ]
     },
     "execution_count": 165,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Nissle_qc_plas['ideal_contig_num'] = Nissle_qc_plas['number_of_plasmids']+1\n",
    "Nissle_qc_plas.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "id": "2215e88e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1508, 12)"
      ]
     },
     "execution_count": 166,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Nissle_qc_plas.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "id": "362126b3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "count    1190.000000\n",
       "mean       99.671324\n",
       "std         0.654338\n",
       "min        91.248000\n",
       "25%        99.805000\n",
       "50%        99.921000\n",
       "75%        99.924000\n",
       "max        99.990000\n",
       "Name: Genome fraction (%), dtype: float64"
      ]
     },
     "execution_count": 172,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Nissle_qc_plas['Genome fraction (%)'].describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "id": "ec80f260",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample-id</th>\n",
       "      <th>Completeness</th>\n",
       "      <th>Contamination</th>\n",
       "      <th># contigs</th>\n",
       "      <th># contigs (&gt;= 50000 bp)</th>\n",
       "      <th>Largest contig</th>\n",
       "      <th>Total length</th>\n",
       "      <th>Reference length</th>\n",
       "      <th>Total aligned length</th>\n",
       "      <th>Genome fraction (%)</th>\n",
       "      <th>number_of_plasmids</th>\n",
       "      <th>ideal_contig_num</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>23B_Day7_12</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5056481.0</td>\n",
       "      <td>5065168.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>5063920.0</td>\n",
       "      <td>99.921</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>23B_Day7_18</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5056492.0</td>\n",
       "      <td>5065179.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>5063931.0</td>\n",
       "      <td>99.921</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>23B_Day7_24</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5056485.0</td>\n",
       "      <td>5065172.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>5063924.0</td>\n",
       "      <td>99.921</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>23B_Day7_5</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5056480.0</td>\n",
       "      <td>5065167.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>5063919.0</td>\n",
       "      <td>99.921</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>23B_Day7_6</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5056392.0</td>\n",
       "      <td>5065079.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>5063843.0</td>\n",
       "      <td>99.920</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      sample-id  Completeness  Contamination  # contigs  \\\n",
       "1   23B_Day7_12         100.0            0.0        3.0   \n",
       "6   23B_Day7_18         100.0            0.0        3.0   \n",
       "11  23B_Day7_24         100.0            0.0        3.0   \n",
       "15   23B_Day7_5         100.0            0.0        3.0   \n",
       "16   23B_Day7_6         100.0            0.0        3.0   \n",
       "\n",
       "    # contigs (>= 50000 bp)  Largest contig  Total length  Reference length  \\\n",
       "1                       1.0       5056481.0     5065168.0         5064668.0   \n",
       "6                       1.0       5056492.0     5065179.0         5064668.0   \n",
       "11                      1.0       5056485.0     5065172.0         5064668.0   \n",
       "15                      1.0       5056480.0     5065167.0         5064668.0   \n",
       "16                      1.0       5056392.0     5065079.0         5064668.0   \n",
       "\n",
       "    Total aligned length  Genome fraction (%)  number_of_plasmids  \\\n",
       "1              5063920.0               99.921                   3   \n",
       "6              5063931.0               99.921                   3   \n",
       "11             5063924.0               99.921                   2   \n",
       "15             5063919.0               99.921                   4   \n",
       "16             5063843.0               99.920                   3   \n",
       "\n",
       "    ideal_contig_num  \n",
       "1                  4  \n",
       "6                  4  \n",
       "11                 3  \n",
       "15                 5  \n",
       "16                 4  "
      ]
     },
     "execution_count": 167,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Nissle_ideal = Nissle_qc_plas[Nissle_qc_plas['# contigs']<=Nissle_qc_plas['ideal_contig_num']]\n",
    "Nissle_ideal.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "id": "5e13e673",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(618, 12)"
      ]
     },
     "execution_count": 168,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Nissle_ideal.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "id": "a94c4546",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample-id</th>\n",
       "      <th>Completeness</th>\n",
       "      <th>Contamination</th>\n",
       "      <th># contigs</th>\n",
       "      <th># contigs (&gt;= 50000 bp)</th>\n",
       "      <th>Largest contig</th>\n",
       "      <th>Total length</th>\n",
       "      <th>Reference length</th>\n",
       "      <th>Total aligned length</th>\n",
       "      <th>Genome fraction (%)</th>\n",
       "      <th>number_of_plasmids</th>\n",
       "      <th>ideal_contig_num</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>23B_Day7_12</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5056481.0</td>\n",
       "      <td>5065168.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>5063920.0</td>\n",
       "      <td>99.921</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>23B_Day7_13</td>\n",
       "      <td>100.0</td>\n",
       "      <td>13.79</td>\n",
       "      <td>31.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5054136.0</td>\n",
       "      <td>5475531.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>5471892.0</td>\n",
       "      <td>99.869</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>23B_Day7_18</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5056492.0</td>\n",
       "      <td>5065179.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>5063931.0</td>\n",
       "      <td>99.921</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>23B_Day7_24</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5056485.0</td>\n",
       "      <td>5065172.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>5063924.0</td>\n",
       "      <td>99.921</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>23B_Day7_2</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>7.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5055457.0</td>\n",
       "      <td>5086402.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>5062181.0</td>\n",
       "      <td>99.883</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1216</th>\n",
       "      <td>30N_Day7_4</td>\n",
       "      <td>100.0</td>\n",
       "      <td>6.80</td>\n",
       "      <td>156.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5053813.0</td>\n",
       "      <td>5487836.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>5482894.0</td>\n",
       "      <td>99.946</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1217</th>\n",
       "      <td>30N_Day7_5</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5056489.0</td>\n",
       "      <td>5065176.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>5063928.0</td>\n",
       "      <td>99.921</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1218</th>\n",
       "      <td>30N_Day7_6</td>\n",
       "      <td>100.0</td>\n",
       "      <td>10.34</td>\n",
       "      <td>64.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5054787.0</td>\n",
       "      <td>5466256.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>5458725.0</td>\n",
       "      <td>99.818</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1219</th>\n",
       "      <td>30N_Day7_7</td>\n",
       "      <td>100.0</td>\n",
       "      <td>2.35</td>\n",
       "      <td>170.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5034469.0</td>\n",
       "      <td>5436994.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>5432914.0</td>\n",
       "      <td>99.623</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1221</th>\n",
       "      <td>30N_Day7_9</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>23.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5054533.0</td>\n",
       "      <td>5289914.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>5285524.0</td>\n",
       "      <td>99.823</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>765 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample-id  Completeness  Contamination  # contigs  \\\n",
       "1     23B_Day7_12         100.0           0.00        3.0   \n",
       "2     23B_Day7_13         100.0          13.79       31.0   \n",
       "6     23B_Day7_18         100.0           0.00        3.0   \n",
       "11    23B_Day7_24         100.0           0.00        3.0   \n",
       "12     23B_Day7_2         100.0           0.00        7.0   \n",
       "...           ...           ...            ...        ...   \n",
       "1216   30N_Day7_4         100.0           6.80      156.0   \n",
       "1217   30N_Day7_5         100.0           0.00        3.0   \n",
       "1218   30N_Day7_6         100.0          10.34       64.0   \n",
       "1219   30N_Day7_7         100.0           2.35      170.0   \n",
       "1221   30N_Day7_9         100.0           0.00       23.0   \n",
       "\n",
       "      # contigs (>= 50000 bp)  Largest contig  Total length  Reference length  \\\n",
       "1                         1.0       5056481.0     5065168.0         5064668.0   \n",
       "2                         1.0       5054136.0     5475531.0         5064668.0   \n",
       "6                         1.0       5056492.0     5065179.0         5064668.0   \n",
       "11                        1.0       5056485.0     5065172.0         5064668.0   \n",
       "12                        1.0       5055457.0     5086402.0         5064668.0   \n",
       "...                       ...             ...           ...               ...   \n",
       "1216                      1.0       5053813.0     5487836.0         5064668.0   \n",
       "1217                      1.0       5056489.0     5065176.0         5064668.0   \n",
       "1218                      1.0       5054787.0     5466256.0         5064668.0   \n",
       "1219                      1.0       5034469.0     5436994.0         5064668.0   \n",
       "1221                      1.0       5054533.0     5289914.0         5064668.0   \n",
       "\n",
       "      Total aligned length  Genome fraction (%)  number_of_plasmids  \\\n",
       "1                5063920.0               99.921                   3   \n",
       "2                5471892.0               99.869                   2   \n",
       "6                5063931.0               99.921                   3   \n",
       "11               5063924.0               99.921                   2   \n",
       "12               5062181.0               99.883                   4   \n",
       "...                    ...                  ...                 ...   \n",
       "1216             5482894.0               99.946                   2   \n",
       "1217             5063928.0               99.921                   3   \n",
       "1218             5458725.0               99.818                   2   \n",
       "1219             5432914.0               99.623                   2   \n",
       "1221             5285524.0               99.823                   2   \n",
       "\n",
       "      ideal_contig_num  \n",
       "1                    4  \n",
       "2                    3  \n",
       "6                    4  \n",
       "11                   3  \n",
       "12                   5  \n",
       "...                ...  \n",
       "1216                 3  \n",
       "1217                 4  \n",
       "1218                 3  \n",
       "1219                 3  \n",
       "1221                 3  \n",
       "\n",
       "[765 rows x 12 columns]"
      ]
     },
     "execution_count": 169,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Nissle_clean = Nissle_qc_plas[(Nissle_qc_plas['Completeness']>99)&(Nissle_qc_plas['# contigs (>= 50000 bp)']<2)&(Nissle_qc_plas['Largest contig']>4500000)]\n",
    "Nissle_clean"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "id": "c2898b42",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(765, 12)"
      ]
     },
     "execution_count": 170,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Nissle_clean.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "id": "3a8cecb8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample-id</th>\n",
       "      <th>Completeness</th>\n",
       "      <th>Contamination</th>\n",
       "      <th># contigs</th>\n",
       "      <th># contigs (&gt;= 50000 bp)</th>\n",
       "      <th>Largest contig</th>\n",
       "      <th>Total length</th>\n",
       "      <th>Reference length</th>\n",
       "      <th>Total aligned length</th>\n",
       "      <th>Genome fraction (%)</th>\n",
       "      <th>number_of_plasmids</th>\n",
       "      <th>ideal_contig_num</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>23B_Day7_12</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5056481.0</td>\n",
       "      <td>5065168.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>5063920.0</td>\n",
       "      <td>99.921</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>23B_Day7_13</td>\n",
       "      <td>100.0</td>\n",
       "      <td>13.79</td>\n",
       "      <td>31.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5054136.0</td>\n",
       "      <td>5475531.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>5471892.0</td>\n",
       "      <td>99.869</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>23B_Day7_14</td>\n",
       "      <td>100.0</td>\n",
       "      <td>5.17</td>\n",
       "      <td>16.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>2787855.0</td>\n",
       "      <td>5363807.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>5360446.0</td>\n",
       "      <td>99.880</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>23B_Day7_16</td>\n",
       "      <td>100.0</td>\n",
       "      <td>12.07</td>\n",
       "      <td>17.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5046245.0</td>\n",
       "      <td>5554755.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>5552302.0</td>\n",
       "      <td>99.776</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>23B_Day7_18</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5056492.0</td>\n",
       "      <td>5065179.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>5063931.0</td>\n",
       "      <td>99.921</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1217</th>\n",
       "      <td>30N_Day7_5</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5056489.0</td>\n",
       "      <td>5065176.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>5063928.0</td>\n",
       "      <td>99.921</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1218</th>\n",
       "      <td>30N_Day7_6</td>\n",
       "      <td>100.0</td>\n",
       "      <td>10.34</td>\n",
       "      <td>64.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5054787.0</td>\n",
       "      <td>5466256.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>5458725.0</td>\n",
       "      <td>99.818</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1219</th>\n",
       "      <td>30N_Day7_7</td>\n",
       "      <td>100.0</td>\n",
       "      <td>2.35</td>\n",
       "      <td>170.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5034469.0</td>\n",
       "      <td>5436994.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>5432914.0</td>\n",
       "      <td>99.623</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1220</th>\n",
       "      <td>30N_Day7_8</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1.72</td>\n",
       "      <td>60.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2167375.0</td>\n",
       "      <td>5272087.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>5270760.0</td>\n",
       "      <td>99.141</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1221</th>\n",
       "      <td>30N_Day7_9</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>23.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5054533.0</td>\n",
       "      <td>5289914.0</td>\n",
       "      <td>5064668.0</td>\n",
       "      <td>5285524.0</td>\n",
       "      <td>99.823</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1070 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample-id  Completeness  Contamination  # contigs  \\\n",
       "1     23B_Day7_12         100.0           0.00        3.0   \n",
       "2     23B_Day7_13         100.0          13.79       31.0   \n",
       "3     23B_Day7_14         100.0           5.17       16.0   \n",
       "4     23B_Day7_16         100.0          12.07       17.0   \n",
       "6     23B_Day7_18         100.0           0.00        3.0   \n",
       "...           ...           ...            ...        ...   \n",
       "1217   30N_Day7_5         100.0           0.00        3.0   \n",
       "1218   30N_Day7_6         100.0          10.34       64.0   \n",
       "1219   30N_Day7_7         100.0           2.35      170.0   \n",
       "1220   30N_Day7_8         100.0           1.72       60.0   \n",
       "1221   30N_Day7_9         100.0           0.00       23.0   \n",
       "\n",
       "      # contigs (>= 50000 bp)  Largest contig  Total length  Reference length  \\\n",
       "1                         1.0       5056481.0     5065168.0         5064668.0   \n",
       "2                         1.0       5054136.0     5475531.0         5064668.0   \n",
       "3                         8.0       2787855.0     5363807.0         5064668.0   \n",
       "4                         4.0       5046245.0     5554755.0         5064668.0   \n",
       "6                         1.0       5056492.0     5065179.0         5064668.0   \n",
       "...                       ...             ...           ...               ...   \n",
       "1217                      1.0       5056489.0     5065176.0         5064668.0   \n",
       "1218                      1.0       5054787.0     5466256.0         5064668.0   \n",
       "1219                      1.0       5034469.0     5436994.0         5064668.0   \n",
       "1220                      6.0       2167375.0     5272087.0         5064668.0   \n",
       "1221                      1.0       5054533.0     5289914.0         5064668.0   \n",
       "\n",
       "      Total aligned length  Genome fraction (%)  number_of_plasmids  \\\n",
       "1                5063920.0               99.921                   3   \n",
       "2                5471892.0               99.869                   2   \n",
       "3                5360446.0               99.880                   2   \n",
       "4                5552302.0               99.776                   2   \n",
       "6                5063931.0               99.921                   3   \n",
       "...                    ...                  ...                 ...   \n",
       "1217             5063928.0               99.921                   3   \n",
       "1218             5458725.0               99.818                   2   \n",
       "1219             5432914.0               99.623                   2   \n",
       "1220             5270760.0               99.141                   2   \n",
       "1221             5285524.0               99.823                   2   \n",
       "\n",
       "      ideal_contig_num  \n",
       "1                    4  \n",
       "2                    3  \n",
       "3                    3  \n",
       "4                    3  \n",
       "6                    4  \n",
       "...                ...  \n",
       "1217                 4  \n",
       "1218                 3  \n",
       "1219                 3  \n",
       "1220                 3  \n",
       "1221                 3  \n",
       "\n",
       "[1070 rows x 12 columns]"
      ]
     },
     "execution_count": 188,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Nissle_best = Nissle_qc_plas[(Nissle_qc_plas['Genome fraction (%)']>99)]\n",
    "#percentage of aligned bases in the reference genome. \n",
    "#A base in the reference genome is aligned if there is at least one contig with at least one alignment to this base.\n",
    "Nissle_best"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "87be26c8",
   "metadata": {},
   "source": [
    "# AZ51"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "b5290de5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Marker lineage</th>\n",
       "      <th># genomes</th>\n",
       "      <th># markers</th>\n",
       "      <th># marker sets</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5+</th>\n",
       "      <th>...</th>\n",
       "      <th>status</th>\n",
       "      <th>superkingdom</th>\n",
       "      <th>phylum</th>\n",
       "      <th>class</th>\n",
       "      <th>order</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "      <th>species</th>\n",
       "      <th>strain</th>\n",
       "      <th>sample_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>41B_Day7_10_S73_L002</th>\n",
       "      <td>Bacteria</td>\n",
       "      <td>5449</td>\n",
       "      <td>104</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "      <td>104</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>found</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales</td>\n",
       "      <td>Enterobacteriaceae</td>\n",
       "      <td>Escherichia</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>41B_Day7_10_S73_L002.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41B_Day7_11_S81_L002</th>\n",
       "      <td>Bacteria</td>\n",
       "      <td>5449</td>\n",
       "      <td>104</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "      <td>104</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>found</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales</td>\n",
       "      <td>Enterobacteriaceae</td>\n",
       "      <td>Escherichia</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>41B_Day7_11_S81_L002.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41B_Day7_12_S89_L002</th>\n",
       "      <td>Bacteria</td>\n",
       "      <td>5449</td>\n",
       "      <td>104</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "      <td>104</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>found</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales</td>\n",
       "      <td>Enterobacteriaceae</td>\n",
       "      <td>Escherichia</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>41B_Day7_12_S89_L002.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41B_Day7_13_S2_L002</th>\n",
       "      <td>Bacteria</td>\n",
       "      <td>5449</td>\n",
       "      <td>104</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "      <td>104</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>found</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales</td>\n",
       "      <td>Enterobacteriaceae</td>\n",
       "      <td>Escherichia</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>41B_Day7_13_S2_L002.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41B_Day7_14_S10_L002</th>\n",
       "      <td>Bacteria</td>\n",
       "      <td>5449</td>\n",
       "      <td>104</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "      <td>104</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>found</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales</td>\n",
       "      <td>Enterobacteriaceae</td>\n",
       "      <td>Escherichia</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>41B_Day7_14_S10_L002.fasta</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 77 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     Marker lineage  # genomes  # markers  # marker sets  0  \\\n",
       "41B_Day7_10_S73_L002       Bacteria       5449        104             58  0   \n",
       "41B_Day7_11_S81_L002       Bacteria       5449        104             58  0   \n",
       "41B_Day7_12_S89_L002       Bacteria       5449        104             58  0   \n",
       "41B_Day7_13_S2_L002        Bacteria       5449        104             58  0   \n",
       "41B_Day7_14_S10_L002       Bacteria       5449        104             58  0   \n",
       "\n",
       "                        1  2  3  4  5+  ...  status  superkingdom  \\\n",
       "41B_Day7_10_S73_L002  104  0  0  0   0  ...   found      Bacteria   \n",
       "41B_Day7_11_S81_L002  104  0  0  0   0  ...   found      Bacteria   \n",
       "41B_Day7_12_S89_L002  104  0  0  0   0  ...   found      Bacteria   \n",
       "41B_Day7_13_S2_L002   104  0  0  0   0  ...   found      Bacteria   \n",
       "41B_Day7_14_S10_L002  104  0  0  0   0  ...   found      Bacteria   \n",
       "\n",
       "                              phylum                class             order  \\\n",
       "41B_Day7_10_S73_L002  Proteobacteria  Gammaproteobacteria  Enterobacterales   \n",
       "41B_Day7_11_S81_L002  Proteobacteria  Gammaproteobacteria  Enterobacterales   \n",
       "41B_Day7_12_S89_L002  Proteobacteria  Gammaproteobacteria  Enterobacterales   \n",
       "41B_Day7_13_S2_L002   Proteobacteria  Gammaproteobacteria  Enterobacterales   \n",
       "41B_Day7_14_S10_L002  Proteobacteria  Gammaproteobacteria  Enterobacterales   \n",
       "\n",
       "                                  family        genus           species  \\\n",
       "41B_Day7_10_S73_L002  Enterobacteriaceae  Escherichia  Escherichia coli   \n",
       "41B_Day7_11_S81_L002  Enterobacteriaceae  Escherichia  Escherichia coli   \n",
       "41B_Day7_12_S89_L002  Enterobacteriaceae  Escherichia  Escherichia coli   \n",
       "41B_Day7_13_S2_L002   Enterobacteriaceae  Escherichia  Escherichia coli   \n",
       "41B_Day7_14_S10_L002  Enterobacteriaceae  Escherichia  Escherichia coli   \n",
       "\n",
       "                      strain                   sample_id  \n",
       "41B_Day7_10_S73_L002     NaN  41B_Day7_10_S73_L002.fasta  \n",
       "41B_Day7_11_S81_L002     NaN  41B_Day7_11_S81_L002.fasta  \n",
       "41B_Day7_12_S89_L002     NaN  41B_Day7_12_S89_L002.fasta  \n",
       "41B_Day7_13_S2_L002      NaN   41B_Day7_13_S2_L002.fasta  \n",
       "41B_Day7_14_S10_L002     NaN  41B_Day7_14_S10_L002.fasta  \n",
       "\n",
       "[5 rows x 77 columns]"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Native_allqc = pd.read_csv('qc_output_files/AZ51_qc_all.csv', index_col=0)\n",
    "Native_allqc.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "719c85a5",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/m1/h11rhh850_s7fjxgxw5wzh_r0000gp/T/ipykernel_51169/3015971477.py:2: FutureWarning: In a future version of pandas all arguments of StringMethods.rsplit except for the argument 'pat' will be keyword-only.\n",
      "  Native_allqc2['sample-id'] = Native_allqc2['index'].str.rsplit('_', 1).str[0]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>Marker lineage</th>\n",
       "      <th># genomes</th>\n",
       "      <th># markers</th>\n",
       "      <th># marker sets</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>...</th>\n",
       "      <th>superkingdom</th>\n",
       "      <th>phylum</th>\n",
       "      <th>class</th>\n",
       "      <th>order</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "      <th>species</th>\n",
       "      <th>strain</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>sample-id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>41B_Day7_10_S73_L002</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>5449</td>\n",
       "      <td>104</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "      <td>104</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales</td>\n",
       "      <td>Enterobacteriaceae</td>\n",
       "      <td>Escherichia</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>41B_Day7_10_S73_L002.fasta</td>\n",
       "      <td>41B_Day7_10_S73</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>41B_Day7_11_S81_L002</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>5449</td>\n",
       "      <td>104</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "      <td>104</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales</td>\n",
       "      <td>Enterobacteriaceae</td>\n",
       "      <td>Escherichia</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>41B_Day7_11_S81_L002.fasta</td>\n",
       "      <td>41B_Day7_11_S81</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>41B_Day7_12_S89_L002</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>5449</td>\n",
       "      <td>104</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "      <td>104</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales</td>\n",
       "      <td>Enterobacteriaceae</td>\n",
       "      <td>Escherichia</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>41B_Day7_12_S89_L002.fasta</td>\n",
       "      <td>41B_Day7_12_S89</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>41B_Day7_13_S2_L002</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>5449</td>\n",
       "      <td>104</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "      <td>104</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales</td>\n",
       "      <td>Enterobacteriaceae</td>\n",
       "      <td>Escherichia</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>41B_Day7_13_S2_L002.fasta</td>\n",
       "      <td>41B_Day7_13_S2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>41B_Day7_14_S10_L002</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>5449</td>\n",
       "      <td>104</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "      <td>104</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales</td>\n",
       "      <td>Enterobacteriaceae</td>\n",
       "      <td>Escherichia</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>41B_Day7_14_S10_L002.fasta</td>\n",
       "      <td>41B_Day7_14_S10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 79 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  index Marker lineage  # genomes  # markers  # marker sets  \\\n",
       "0  41B_Day7_10_S73_L002       Bacteria       5449        104             58   \n",
       "1  41B_Day7_11_S81_L002       Bacteria       5449        104             58   \n",
       "2  41B_Day7_12_S89_L002       Bacteria       5449        104             58   \n",
       "3   41B_Day7_13_S2_L002       Bacteria       5449        104             58   \n",
       "4  41B_Day7_14_S10_L002       Bacteria       5449        104             58   \n",
       "\n",
       "   0    1  2  3  4  ...  superkingdom          phylum                class  \\\n",
       "0  0  104  0  0  0  ...      Bacteria  Proteobacteria  Gammaproteobacteria   \n",
       "1  0  104  0  0  0  ...      Bacteria  Proteobacteria  Gammaproteobacteria   \n",
       "2  0  104  0  0  0  ...      Bacteria  Proteobacteria  Gammaproteobacteria   \n",
       "3  0  104  0  0  0  ...      Bacteria  Proteobacteria  Gammaproteobacteria   \n",
       "4  0  104  0  0  0  ...      Bacteria  Proteobacteria  Gammaproteobacteria   \n",
       "\n",
       "              order              family        genus           species  \\\n",
       "0  Enterobacterales  Enterobacteriaceae  Escherichia  Escherichia coli   \n",
       "1  Enterobacterales  Enterobacteriaceae  Escherichia  Escherichia coli   \n",
       "2  Enterobacterales  Enterobacteriaceae  Escherichia  Escherichia coli   \n",
       "3  Enterobacterales  Enterobacteriaceae  Escherichia  Escherichia coli   \n",
       "4  Enterobacterales  Enterobacteriaceae  Escherichia  Escherichia coli   \n",
       "\n",
       "   strain                   sample_id        sample-id  \n",
       "0     NaN  41B_Day7_10_S73_L002.fasta  41B_Day7_10_S73  \n",
       "1     NaN  41B_Day7_11_S81_L002.fasta  41B_Day7_11_S81  \n",
       "2     NaN  41B_Day7_12_S89_L002.fasta  41B_Day7_12_S89  \n",
       "3     NaN   41B_Day7_13_S2_L002.fasta   41B_Day7_13_S2  \n",
       "4     NaN  41B_Day7_14_S10_L002.fasta  41B_Day7_14_S10  \n",
       "\n",
       "[5 rows x 79 columns]"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Native_allqc2 = Native_allqc.reset_index()\n",
    "Native_allqc2['sample-id'] = Native_allqc2['index'].str.rsplit('_', 1).str[0]\n",
    "Native_allqc2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "01fcdb62",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['index', 'Marker lineage', '# genomes', '# markers', '# marker sets',\n",
       "       '0', '1', '2', '3', '4', '5+', 'Completeness', 'Contamination',\n",
       "       'Strain heterogeneity', '# unique markers (of 43)', '# multi-copy',\n",
       "       'Taxonomy', '# contigs (>= 0 bp)', '# contigs (>= 1000 bp)',\n",
       "       '# contigs (>= 5000 bp)', '# contigs (>= 10000 bp)',\n",
       "       '# contigs (>= 25000 bp)', '# contigs (>= 50000 bp)',\n",
       "       'Total length (>= 0 bp)', 'Total length (>= 1000 bp)',\n",
       "       'Total length (>= 5000 bp)', 'Total length (>= 10000 bp)',\n",
       "       'Total length (>= 25000 bp)', 'Total length (>= 50000 bp)', '# contigs',\n",
       "       'Largest contig', 'Total length', 'Reference length', 'GC (%)',\n",
       "       'Reference GC (%)', 'N50', 'NG50', 'N75', 'NG75', 'L50', 'LG50', 'L75',\n",
       "       'LG75', '# misassemblies', '# misassembled contigs',\n",
       "       'Misassembled contigs length', '# local misassemblies',\n",
       "       '# scaffold gap ext. mis.', '# scaffold gap loc. mis.',\n",
       "       '# unaligned mis. contigs', '# unaligned contigs', 'Unaligned length',\n",
       "       'Genome fraction (%)', 'Duplication ratio', '# N's per 100 kbp',\n",
       "       '# mismatches per 100 kbp', '# indels per 100 kbp',\n",
       "       '# genomic features', 'Largest alignment', 'Total aligned length',\n",
       "       'NA50', 'NGA50', 'NA75', 'NGA75', 'LA50', 'LGA50', 'LA75', 'LGA75',\n",
       "       'status', 'superkingdom', 'phylum', 'class', 'order', 'family', 'genus',\n",
       "       'species', 'strain', 'sample_id', 'sample-id'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Native_allqc2.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "b9db41a8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample-id</th>\n",
       "      <th>Completeness</th>\n",
       "      <th>Contamination</th>\n",
       "      <th># contigs</th>\n",
       "      <th># contigs (&gt;= 50000 bp)</th>\n",
       "      <th>Largest contig</th>\n",
       "      <th>Total length</th>\n",
       "      <th>Reference length</th>\n",
       "      <th>Total aligned length</th>\n",
       "      <th>Genome fraction (%)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>41B_Day7_10_S73</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>5011504</td>\n",
       "      <td>5061325</td>\n",
       "      <td>5062596</td>\n",
       "      <td>5061325</td>\n",
       "      <td>99.975</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>41B_Day7_11_S81</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>5011504</td>\n",
       "      <td>5061756</td>\n",
       "      <td>5062596</td>\n",
       "      <td>5061756</td>\n",
       "      <td>99.983</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>41B_Day7_12_S89</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>39</td>\n",
       "      <td>13</td>\n",
       "      <td>1185431</td>\n",
       "      <td>5011916</td>\n",
       "      <td>5062596</td>\n",
       "      <td>5011916</td>\n",
       "      <td>98.999</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>41B_Day7_13_S2</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>37</td>\n",
       "      <td>12</td>\n",
       "      <td>1185431</td>\n",
       "      <td>5013543</td>\n",
       "      <td>5062596</td>\n",
       "      <td>5013543</td>\n",
       "      <td>99.031</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>41B_Day7_14_S10</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>37</td>\n",
       "      <td>12</td>\n",
       "      <td>1185431</td>\n",
       "      <td>5013542</td>\n",
       "      <td>5062596</td>\n",
       "      <td>5013542</td>\n",
       "      <td>99.031</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         sample-id  Completeness  Contamination  # contigs  \\\n",
       "0  41B_Day7_10_S73         100.0            0.0          5   \n",
       "1  41B_Day7_11_S81         100.0            0.0          4   \n",
       "2  41B_Day7_12_S89         100.0            0.0         39   \n",
       "3   41B_Day7_13_S2         100.0            0.0         37   \n",
       "4  41B_Day7_14_S10         100.0            0.0         37   \n",
       "\n",
       "   # contigs (>= 50000 bp)  Largest contig  Total length  Reference length  \\\n",
       "0                        1         5011504       5061325           5062596   \n",
       "1                        1         5011504       5061756           5062596   \n",
       "2                       13         1185431       5011916           5062596   \n",
       "3                       12         1185431       5013543           5062596   \n",
       "4                       12         1185431       5013542           5062596   \n",
       "\n",
       "   Total aligned length  Genome fraction (%)  \n",
       "0               5061325               99.975  \n",
       "1               5061756               99.983  \n",
       "2               5011916               98.999  \n",
       "3               5013543               99.031  \n",
       "4               5013542               99.031  "
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Native_primqc = Native_allqc2[['sample-id', 'Completeness', 'Contamination', \n",
    "                              '# contigs', '# contigs (>= 50000 bp)', 'Largest contig', \n",
    "                              'Total length', 'Reference length', 'Total aligned length',\n",
    "                              'Genome fraction (%)']].copy()\n",
    "Native_primqc.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "ebdcef8c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(2023, 10)"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Native_primqc.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "1e383416",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>41B_Day7_10_S73/large_plasmids.fasta:2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>41B_Day7_11_S81/large_plasmids.fasta:1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>41B_Day7_12_S89/large_plasmids.fasta:2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>41B_Day7_13_S2/large_plasmids.fasta:1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>41B_Day7_14_S10/large_plasmids.fasta:2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1964</th>\n",
       "      <td>46R_Month6_6_S717/large_plasmids.fasta:3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1965</th>\n",
       "      <td>46R_Month6_7_S725/large_plasmids.fasta:3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1966</th>\n",
       "      <td>46R_Month6_8_S733/large_plasmids.fasta:3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1967</th>\n",
       "      <td>46R_Month6_9_S741/large_plasmids.fasta:2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1968</th>\n",
       "      <td>AZ51_ref/large_plasmids.fasta:2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1969 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             0\n",
       "0       41B_Day7_10_S73/large_plasmids.fasta:2\n",
       "1       41B_Day7_11_S81/large_plasmids.fasta:1\n",
       "2       41B_Day7_12_S89/large_plasmids.fasta:2\n",
       "3        41B_Day7_13_S2/large_plasmids.fasta:1\n",
       "4       41B_Day7_14_S10/large_plasmids.fasta:2\n",
       "...                                        ...\n",
       "1964  46R_Month6_6_S717/large_plasmids.fasta:3\n",
       "1965  46R_Month6_7_S725/large_plasmids.fasta:3\n",
       "1966  46R_Month6_8_S733/large_plasmids.fasta:3\n",
       "1967  46R_Month6_9_S741/large_plasmids.fasta:2\n",
       "1968           AZ51_ref/large_plasmids.fasta:2\n",
       "\n",
       "[1969 rows x 1 columns]"
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Native_plasmid_counts = pd.read_csv('../plasmids/AZ51_lplasmid_number.txt', sep='\\t', header=None)\n",
    "Native_plasmid_counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "a637a3a6",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/m1/h11rhh850_s7fjxgxw5wzh_r0000gp/T/ipykernel_51169/134096259.py:1: FutureWarning: In a future version of pandas all arguments of StringMethods.split except for the argument 'pat' will be keyword-only.\n",
      "  Native_plasmid_counts[['sample', 'other']] = Native_plasmid_counts[0].str.split('/', 1, expand=True)\n"
     ]
    }
   ],
   "source": [
    "Native_plasmid_counts[['sample', 'other']] = Native_plasmid_counts[0].str.split('/', 1, expand=True)\n",
    "Native_plasmid_counts['number_of_plasmids'] = Native_plasmid_counts['other'].str.split(':').str[1]\n",
    "Native_plasmid_counts['sample-id'] = Native_plasmid_counts['sample'].replace(to_replace='_S[0-9]*', value='', regex=True)\n",
    "Native_plasmid_counts['sample-id'] = Native_plasmid_counts['sample-id'].replace(to_replace='_L003', value='', regex=True)\n",
    "Native_plasmid_counts['sample-id'] = Native_plasmid_counts['sample-id'].replace(to_replace='_L002', value='', regex=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "9c85a83f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample-id</th>\n",
       "      <th>number_of_plasmids</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>41B_Day7_10</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>41B_Day7_11</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>41B_Day7_12</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>41B_Day7_13</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>41B_Day7_14</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     sample-id number_of_plasmids\n",
       "0  41B_Day7_10                  2\n",
       "1  41B_Day7_11                  1\n",
       "2  41B_Day7_12                  2\n",
       "3  41B_Day7_13                  1\n",
       "4  41B_Day7_14                  2"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Native_plasmid_counts2 = Native_plasmid_counts[['sample-id', 'number_of_plasmids']].copy()\n",
    "Native_plasmid_counts2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "52832548",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample-id</th>\n",
       "      <th>Completeness</th>\n",
       "      <th>Contamination</th>\n",
       "      <th># contigs</th>\n",
       "      <th># contigs (&gt;= 50000 bp)</th>\n",
       "      <th>Largest contig</th>\n",
       "      <th>Total length</th>\n",
       "      <th>Reference length</th>\n",
       "      <th>Total aligned length</th>\n",
       "      <th>Genome fraction (%)</th>\n",
       "      <th>number_of_plasmids</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>41B_Day7_10_S73</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5011504.0</td>\n",
       "      <td>5061325.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5061325.0</td>\n",
       "      <td>99.975</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>41B_Day7_11_S81</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5011504.0</td>\n",
       "      <td>5061756.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5061756.0</td>\n",
       "      <td>99.983</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>41B_Day7_12_S89</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>39.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>1185431.0</td>\n",
       "      <td>5011916.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5011916.0</td>\n",
       "      <td>98.999</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>41B_Day7_13_S2</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>1185431.0</td>\n",
       "      <td>5013543.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5013543.0</td>\n",
       "      <td>99.031</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>41B_Day7_14_S10</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>1185431.0</td>\n",
       "      <td>5013542.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5013542.0</td>\n",
       "      <td>99.031</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2658</th>\n",
       "      <td>46R_Day7_7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2659</th>\n",
       "      <td>46R_Day7_8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2660</th>\n",
       "      <td>46R_Day7_9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2661</th>\n",
       "      <td>46R_Month1_13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2662</th>\n",
       "      <td>AZ51_ref</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2663 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            sample-id  Completeness  Contamination  # contigs  \\\n",
       "0     41B_Day7_10_S73         100.0            0.0        5.0   \n",
       "1     41B_Day7_11_S81         100.0            0.0        4.0   \n",
       "2     41B_Day7_12_S89         100.0            0.0       39.0   \n",
       "3      41B_Day7_13_S2         100.0            0.0       37.0   \n",
       "4     41B_Day7_14_S10         100.0            0.0       37.0   \n",
       "...               ...           ...            ...        ...   \n",
       "2658       46R_Day7_7           NaN            NaN        NaN   \n",
       "2659       46R_Day7_8           NaN            NaN        NaN   \n",
       "2660       46R_Day7_9           NaN            NaN        NaN   \n",
       "2661    46R_Month1_13           NaN            NaN        NaN   \n",
       "2662         AZ51_ref           NaN            NaN        NaN   \n",
       "\n",
       "      # contigs (>= 50000 bp)  Largest contig  Total length  Reference length  \\\n",
       "0                         1.0       5011504.0     5061325.0         5062596.0   \n",
       "1                         1.0       5011504.0     5061756.0         5062596.0   \n",
       "2                        13.0       1185431.0     5011916.0         5062596.0   \n",
       "3                        12.0       1185431.0     5013543.0         5062596.0   \n",
       "4                        12.0       1185431.0     5013542.0         5062596.0   \n",
       "...                       ...             ...           ...               ...   \n",
       "2658                      NaN             NaN           NaN               NaN   \n",
       "2659                      NaN             NaN           NaN               NaN   \n",
       "2660                      NaN             NaN           NaN               NaN   \n",
       "2661                      NaN             NaN           NaN               NaN   \n",
       "2662                      NaN             NaN           NaN               NaN   \n",
       "\n",
       "      Total aligned length  Genome fraction (%) number_of_plasmids  \n",
       "0                5061325.0               99.975                NaN  \n",
       "1                5061756.0               99.983                NaN  \n",
       "2                5011916.0               98.999                NaN  \n",
       "3                5013543.0               99.031                NaN  \n",
       "4                5013542.0               99.031                NaN  \n",
       "...                    ...                  ...                ...  \n",
       "2658                   NaN                  NaN                  2  \n",
       "2659                   NaN                  NaN                  1  \n",
       "2660                   NaN                  NaN                  2  \n",
       "2661                   NaN                  NaN                  0  \n",
       "2662                   NaN                  NaN                  2  \n",
       "\n",
       "[2663 rows x 11 columns]"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Native_qc_plas = pd.merge(Native_primqc, Native_plasmid_counts2, how='outer', \n",
    "                          left_on='sample-id', right_on='sample-id')\n",
    "Native_qc_plas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "id": "fbc9d59e",
   "metadata": {},
   "outputs": [],
   "source": [
    "Native_removal = pd.read_csv('../AZ51_bad_genomes.txt')\n",
    "Native_bad = Native_removal.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "id": "2192440f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(2598, 12)"
      ]
     },
     "execution_count": 174,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Native_qc_plas = Native_qc_plas[~(Native_qc_plas['sample-id'].isin(Native_bad))].copy()\n",
    "Native_qc_plas.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "id": "3d23b434",
   "metadata": {},
   "outputs": [],
   "source": [
    "Native_qc_plas['number_of_plasmids'] = Native_qc_plas['number_of_plasmids'].fillna(0)\n",
    "#makes sure to account for everything that exists, even when it might not have detected plasmids\n",
    "Native_qc_plas['number_of_plasmids'] = Native_qc_plas['number_of_plasmids'].astype(int)\n",
    "#values treated as numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "id": "82152ffb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample-id</th>\n",
       "      <th>Completeness</th>\n",
       "      <th>Contamination</th>\n",
       "      <th># contigs</th>\n",
       "      <th># contigs (&gt;= 50000 bp)</th>\n",
       "      <th>Largest contig</th>\n",
       "      <th>Total length</th>\n",
       "      <th>Reference length</th>\n",
       "      <th>Total aligned length</th>\n",
       "      <th>Genome fraction (%)</th>\n",
       "      <th>number_of_plasmids</th>\n",
       "      <th>ideal_contig_num</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>41B_Day7_10_S73</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5011504.0</td>\n",
       "      <td>5061325.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5061325.0</td>\n",
       "      <td>99.975</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>41B_Day7_11_S81</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5011504.0</td>\n",
       "      <td>5061756.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5061756.0</td>\n",
       "      <td>99.983</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>41B_Day7_12_S89</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>39.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>1185431.0</td>\n",
       "      <td>5011916.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5011916.0</td>\n",
       "      <td>98.999</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>41B_Day7_13_S2</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>1185431.0</td>\n",
       "      <td>5013543.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5013543.0</td>\n",
       "      <td>99.031</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>41B_Day7_14_S10</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>1185431.0</td>\n",
       "      <td>5013542.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5013542.0</td>\n",
       "      <td>99.031</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         sample-id  Completeness  Contamination  # contigs  \\\n",
       "0  41B_Day7_10_S73         100.0            0.0        5.0   \n",
       "1  41B_Day7_11_S81         100.0            0.0        4.0   \n",
       "2  41B_Day7_12_S89         100.0            0.0       39.0   \n",
       "3   41B_Day7_13_S2         100.0            0.0       37.0   \n",
       "4  41B_Day7_14_S10         100.0            0.0       37.0   \n",
       "\n",
       "   # contigs (>= 50000 bp)  Largest contig  Total length  Reference length  \\\n",
       "0                      1.0       5011504.0     5061325.0         5062596.0   \n",
       "1                      1.0       5011504.0     5061756.0         5062596.0   \n",
       "2                     13.0       1185431.0     5011916.0         5062596.0   \n",
       "3                     12.0       1185431.0     5013543.0         5062596.0   \n",
       "4                     12.0       1185431.0     5013542.0         5062596.0   \n",
       "\n",
       "   Total aligned length  Genome fraction (%)  number_of_plasmids  \\\n",
       "0             5061325.0               99.975                   0   \n",
       "1             5061756.0               99.983                   0   \n",
       "2             5011916.0               98.999                   0   \n",
       "3             5013543.0               99.031                   0   \n",
       "4             5013542.0               99.031                   0   \n",
       "\n",
       "   ideal_contig_num  \n",
       "0                 1  \n",
       "1                 1  \n",
       "2                 1  \n",
       "3                 1  \n",
       "4                 1  "
      ]
     },
     "execution_count": 176,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Native_qc_plas['ideal_contig_num'] = Native_qc_plas['number_of_plasmids']+1\n",
    "Native_qc_plas.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "id": "65bb6cf5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(2598, 12)"
      ]
     },
     "execution_count": 177,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Native_qc_plas.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 183,
   "id": "cff73cfb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "count    1960.000000\n",
       "mean       99.510942\n",
       "std         0.641944\n",
       "min        92.340000\n",
       "25%        99.027000\n",
       "50%        99.865000\n",
       "75%        99.978000\n",
       "max       100.000000\n",
       "Name: Genome fraction (%), dtype: float64"
      ]
     },
     "execution_count": 183,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Native_qc_plas['Genome fraction (%)'].describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "id": "24125bcf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample-id</th>\n",
       "      <th>Completeness</th>\n",
       "      <th>Contamination</th>\n",
       "      <th># contigs</th>\n",
       "      <th># contigs (&gt;= 50000 bp)</th>\n",
       "      <th>Largest contig</th>\n",
       "      <th>Total length</th>\n",
       "      <th>Reference length</th>\n",
       "      <th>Total aligned length</th>\n",
       "      <th>Genome fraction (%)</th>\n",
       "      <th>number_of_plasmids</th>\n",
       "      <th>ideal_contig_num</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>41B_Month1_13</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5011503.0</td>\n",
       "      <td>5061755.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5061509.0</td>\n",
       "      <td>99.979</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>41B_Month1_1</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5011504.0</td>\n",
       "      <td>5061325.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5061029.0</td>\n",
       "      <td>99.969</td>\n",
       "      <td>12</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>41B_Month1_20</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5011504.0</td>\n",
       "      <td>5061756.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5061756.0</td>\n",
       "      <td>99.983</td>\n",
       "      <td>12</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>41B_Month1_2</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5011504.0</td>\n",
       "      <td>5061756.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5061756.0</td>\n",
       "      <td>99.983</td>\n",
       "      <td>10</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>41B_Month1_7</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5011504.0</td>\n",
       "      <td>5061093.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5061093.0</td>\n",
       "      <td>99.970</td>\n",
       "      <td>11</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample-id  Completeness  Contamination  # contigs  \\\n",
       "27  41B_Month1_13         100.0            0.0        4.0   \n",
       "34   41B_Month1_1         100.0            0.0        5.0   \n",
       "35  41B_Month1_20         100.0            0.0        4.0   \n",
       "40   41B_Month1_2         100.0            0.0        4.0   \n",
       "45   41B_Month1_7         100.0            0.0        7.0   \n",
       "\n",
       "    # contigs (>= 50000 bp)  Largest contig  Total length  Reference length  \\\n",
       "27                      1.0       5011503.0     5061755.0         5062596.0   \n",
       "34                      1.0       5011504.0     5061325.0         5062596.0   \n",
       "35                      1.0       5011504.0     5061756.0         5062596.0   \n",
       "40                      1.0       5011504.0     5061756.0         5062596.0   \n",
       "45                      1.0       5011504.0     5061093.0         5062596.0   \n",
       "\n",
       "    Total aligned length  Genome fraction (%)  number_of_plasmids  \\\n",
       "27             5061509.0               99.979                   5   \n",
       "34             5061029.0               99.969                  12   \n",
       "35             5061756.0               99.983                  12   \n",
       "40             5061756.0               99.983                  10   \n",
       "45             5061093.0               99.970                  11   \n",
       "\n",
       "    ideal_contig_num  \n",
       "27                 6  \n",
       "34                13  \n",
       "35                13  \n",
       "40                11  \n",
       "45                12  "
      ]
     },
     "execution_count": 178,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Native_ideal = Native_qc_plas[Native_qc_plas['# contigs']<=Native_qc_plas['ideal_contig_num']]\n",
    "Native_ideal.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 179,
   "id": "7347b68c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(180, 12)"
      ]
     },
     "execution_count": 179,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Native_ideal.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "id": "14ca974d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample-id</th>\n",
       "      <th>Completeness</th>\n",
       "      <th>Contamination</th>\n",
       "      <th># contigs</th>\n",
       "      <th># contigs (&gt;= 50000 bp)</th>\n",
       "      <th>Largest contig</th>\n",
       "      <th>Total length</th>\n",
       "      <th>Reference length</th>\n",
       "      <th>Total aligned length</th>\n",
       "      <th>Genome fraction (%)</th>\n",
       "      <th>number_of_plasmids</th>\n",
       "      <th>ideal_contig_num</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>41B_Day7_10_S73</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5011504.0</td>\n",
       "      <td>5061325.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5061325.0</td>\n",
       "      <td>99.975</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>41B_Day7_11_S81</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5011504.0</td>\n",
       "      <td>5061756.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5061756.0</td>\n",
       "      <td>99.983</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>41B_Day7_15_S18</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5011504.0</td>\n",
       "      <td>5061756.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5061756.0</td>\n",
       "      <td>99.983</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>41B_Day7_21_S66</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5011504.0</td>\n",
       "      <td>5061756.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5061756.0</td>\n",
       "      <td>99.983</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>41B_Day7_24_S90</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5011505.0</td>\n",
       "      <td>5061757.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5061757.0</td>\n",
       "      <td>99.983</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2017</th>\n",
       "      <td>46R_Month6_4</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5011481.0</td>\n",
       "      <td>5061733.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5061437.0</td>\n",
       "      <td>99.978</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2019</th>\n",
       "      <td>46R_Month6_6</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5011484.0</td>\n",
       "      <td>5061736.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5061736.0</td>\n",
       "      <td>99.983</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2020</th>\n",
       "      <td>46R_Month6_7</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5011496.0</td>\n",
       "      <td>5061748.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5061748.0</td>\n",
       "      <td>99.983</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2021</th>\n",
       "      <td>46R_Month6_8</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5004179.0</td>\n",
       "      <td>5054000.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5051670.0</td>\n",
       "      <td>99.824</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2022</th>\n",
       "      <td>46R_Month6_9</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5011385.0</td>\n",
       "      <td>5061206.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5060910.0</td>\n",
       "      <td>99.967</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1001 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            sample-id  Completeness  Contamination  # contigs  \\\n",
       "0     41B_Day7_10_S73         100.0            0.0        5.0   \n",
       "1     41B_Day7_11_S81         100.0            0.0        4.0   \n",
       "5     41B_Day7_15_S18         100.0            0.0        4.0   \n",
       "12    41B_Day7_21_S66         100.0            0.0        4.0   \n",
       "15    41B_Day7_24_S90         100.0            0.0        4.0   \n",
       "...               ...           ...            ...        ...   \n",
       "2017     46R_Month6_4         100.0            0.0        4.0   \n",
       "2019     46R_Month6_6         100.0            0.0        4.0   \n",
       "2020     46R_Month6_7         100.0            0.0        4.0   \n",
       "2021     46R_Month6_8         100.0            0.0        5.0   \n",
       "2022     46R_Month6_9         100.0            0.0        5.0   \n",
       "\n",
       "      # contigs (>= 50000 bp)  Largest contig  Total length  Reference length  \\\n",
       "0                         1.0       5011504.0     5061325.0         5062596.0   \n",
       "1                         1.0       5011504.0     5061756.0         5062596.0   \n",
       "5                         1.0       5011504.0     5061756.0         5062596.0   \n",
       "12                        1.0       5011504.0     5061756.0         5062596.0   \n",
       "15                        1.0       5011505.0     5061757.0         5062596.0   \n",
       "...                       ...             ...           ...               ...   \n",
       "2017                      1.0       5011481.0     5061733.0         5062596.0   \n",
       "2019                      1.0       5011484.0     5061736.0         5062596.0   \n",
       "2020                      1.0       5011496.0     5061748.0         5062596.0   \n",
       "2021                      1.0       5004179.0     5054000.0         5062596.0   \n",
       "2022                      1.0       5011385.0     5061206.0         5062596.0   \n",
       "\n",
       "      Total aligned length  Genome fraction (%)  number_of_plasmids  \\\n",
       "0                5061325.0               99.975                   0   \n",
       "1                5061756.0               99.983                   0   \n",
       "5                5061756.0               99.983                   0   \n",
       "12               5061756.0               99.983                   0   \n",
       "15               5061757.0               99.983                   0   \n",
       "...                    ...                  ...                 ...   \n",
       "2017             5061437.0               99.978                   2   \n",
       "2019             5061736.0               99.983                   3   \n",
       "2020             5061748.0               99.983                   3   \n",
       "2021             5051670.0               99.824                   3   \n",
       "2022             5060910.0               99.967                   2   \n",
       "\n",
       "      ideal_contig_num  \n",
       "0                    1  \n",
       "1                    1  \n",
       "5                    1  \n",
       "12                   1  \n",
       "15                   1  \n",
       "...                ...  \n",
       "2017                 3  \n",
       "2019                 4  \n",
       "2020                 4  \n",
       "2021                 4  \n",
       "2022                 3  \n",
       "\n",
       "[1001 rows x 12 columns]"
      ]
     },
     "execution_count": 180,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Native_clean = Native_qc_plas[(Native_qc_plas['Completeness']>99)&(Native_qc_plas['# contigs (>= 50000 bp)']<2)&(Native_qc_plas['Largest contig']>4500000)]\n",
    "Native_clean"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "id": "99b7d47b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1001, 12)"
      ]
     },
     "execution_count": 181,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Native_clean.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "id": "b4cdb910",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample-id</th>\n",
       "      <th>Completeness</th>\n",
       "      <th>Contamination</th>\n",
       "      <th># contigs</th>\n",
       "      <th># contigs (&gt;= 50000 bp)</th>\n",
       "      <th>Largest contig</th>\n",
       "      <th>Total length</th>\n",
       "      <th>Reference length</th>\n",
       "      <th>Total aligned length</th>\n",
       "      <th>Genome fraction (%)</th>\n",
       "      <th>number_of_plasmids</th>\n",
       "      <th>ideal_contig_num</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>41B_Day7_10_S73</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5011504.0</td>\n",
       "      <td>5061325.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5061325.0</td>\n",
       "      <td>99.975</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>41B_Day7_11_S81</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5011504.0</td>\n",
       "      <td>5061756.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5061756.0</td>\n",
       "      <td>99.983</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>41B_Day7_13_S2</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>1185431.0</td>\n",
       "      <td>5013543.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5013543.0</td>\n",
       "      <td>99.031</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>41B_Day7_14_S10</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>1185431.0</td>\n",
       "      <td>5013542.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5013542.0</td>\n",
       "      <td>99.031</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>41B_Day7_15_S18</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5011504.0</td>\n",
       "      <td>5061756.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5061756.0</td>\n",
       "      <td>99.983</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2017</th>\n",
       "      <td>46R_Month6_4</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5011481.0</td>\n",
       "      <td>5061733.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5061437.0</td>\n",
       "      <td>99.978</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2019</th>\n",
       "      <td>46R_Month6_6</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5011484.0</td>\n",
       "      <td>5061736.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5061736.0</td>\n",
       "      <td>99.983</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2020</th>\n",
       "      <td>46R_Month6_7</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5011496.0</td>\n",
       "      <td>5061748.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5061748.0</td>\n",
       "      <td>99.983</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2021</th>\n",
       "      <td>46R_Month6_8</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5004179.0</td>\n",
       "      <td>5054000.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5051670.0</td>\n",
       "      <td>99.824</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2022</th>\n",
       "      <td>46R_Month6_9</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5011385.0</td>\n",
       "      <td>5061206.0</td>\n",
       "      <td>5062596.0</td>\n",
       "      <td>5060910.0</td>\n",
       "      <td>99.967</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1696 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            sample-id  Completeness  Contamination  # contigs  \\\n",
       "0     41B_Day7_10_S73         100.0            0.0        5.0   \n",
       "1     41B_Day7_11_S81         100.0            0.0        4.0   \n",
       "3      41B_Day7_13_S2         100.0            0.0       37.0   \n",
       "4     41B_Day7_14_S10         100.0            0.0       37.0   \n",
       "5     41B_Day7_15_S18         100.0            0.0        4.0   \n",
       "...               ...           ...            ...        ...   \n",
       "2017     46R_Month6_4         100.0            0.0        4.0   \n",
       "2019     46R_Month6_6         100.0            0.0        4.0   \n",
       "2020     46R_Month6_7         100.0            0.0        4.0   \n",
       "2021     46R_Month6_8         100.0            0.0        5.0   \n",
       "2022     46R_Month6_9         100.0            0.0        5.0   \n",
       "\n",
       "      # contigs (>= 50000 bp)  Largest contig  Total length  Reference length  \\\n",
       "0                         1.0       5011504.0     5061325.0         5062596.0   \n",
       "1                         1.0       5011504.0     5061756.0         5062596.0   \n",
       "3                        12.0       1185431.0     5013543.0         5062596.0   \n",
       "4                        12.0       1185431.0     5013542.0         5062596.0   \n",
       "5                         1.0       5011504.0     5061756.0         5062596.0   \n",
       "...                       ...             ...           ...               ...   \n",
       "2017                      1.0       5011481.0     5061733.0         5062596.0   \n",
       "2019                      1.0       5011484.0     5061736.0         5062596.0   \n",
       "2020                      1.0       5011496.0     5061748.0         5062596.0   \n",
       "2021                      1.0       5004179.0     5054000.0         5062596.0   \n",
       "2022                      1.0       5011385.0     5061206.0         5062596.0   \n",
       "\n",
       "      Total aligned length  Genome fraction (%)  number_of_plasmids  \\\n",
       "0                5061325.0               99.975                   0   \n",
       "1                5061756.0               99.983                   0   \n",
       "3                5013543.0               99.031                   0   \n",
       "4                5013542.0               99.031                   0   \n",
       "5                5061756.0               99.983                   0   \n",
       "...                    ...                  ...                 ...   \n",
       "2017             5061437.0               99.978                   2   \n",
       "2019             5061736.0               99.983                   3   \n",
       "2020             5061748.0               99.983                   3   \n",
       "2021             5051670.0               99.824                   3   \n",
       "2022             5060910.0               99.967                   2   \n",
       "\n",
       "      ideal_contig_num  \n",
       "0                    1  \n",
       "1                    1  \n",
       "3                    1  \n",
       "4                    1  \n",
       "5                    1  \n",
       "...                ...  \n",
       "2017                 3  \n",
       "2019                 4  \n",
       "2020                 4  \n",
       "2021                 4  \n",
       "2022                 3  \n",
       "\n",
       "[1696 rows x 12 columns]"
      ]
     },
     "execution_count": 187,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Native_best = Native_qc_plas[(Native_qc_plas['Genome fraction (%)']>99)]\n",
    "#percentage of aligned bases in the reference genome. \n",
    "#A base in the reference genome is aligned if there is at least one contig with at least one alignment to this base.\n",
    "Native_best"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9af84a53",
   "metadata": {},
   "source": [
    "# Filter for Coverage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "id": "031839bb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>orig_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>23B_Day7_10_S382_sorted.bam</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>23B_Day7_11_S383_sorted.bam</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>23B_Day7_12_S384_sorted.bam</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>23B_Day7_13_S1294_sorted.bam</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>23B_Day7_14_S1295_sorted.bam</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1552</th>\n",
       "      <td>30N_Day7_5_S923_sorted.bam</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1553</th>\n",
       "      <td>30N_Day7_6_S924_sorted.bam</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1554</th>\n",
       "      <td>30N_Day7_7_S925_sorted.bam</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1555</th>\n",
       "      <td>30N_Day7_8_S926_sorted.bam</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1556</th>\n",
       "      <td>30N_Day7_9_S927_sorted.bam</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1557 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                           orig_id\n",
       "0      23B_Day7_10_S382_sorted.bam\n",
       "1      23B_Day7_11_S383_sorted.bam\n",
       "2      23B_Day7_12_S384_sorted.bam\n",
       "3     23B_Day7_13_S1294_sorted.bam\n",
       "4     23B_Day7_14_S1295_sorted.bam\n",
       "...                            ...\n",
       "1552    30N_Day7_5_S923_sorted.bam\n",
       "1553    30N_Day7_6_S924_sorted.bam\n",
       "1554    30N_Day7_7_S925_sorted.bam\n",
       "1555    30N_Day7_8_S926_sorted.bam\n",
       "1556    30N_Day7_9_S927_sorted.bam\n",
       "\n",
       "[1557 rows x 1 columns]"
      ]
     },
     "execution_count": 208,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Nissle_sortedbam = pd.read_csv('AZ20_sortedbam_list_full.txt', sep='\\t', header=None)\n",
    "Nissle_sortedbam = pd.DataFrame(Nissle_sortedbam)\n",
    "Nissle_sortedbam = Nissle_sortedbam.rename(columns={0:'orig_id'})\n",
    "Nissle_sortedbam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "id": "6110c25c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>orig_id</th>\n",
       "      <th>sample-id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>23B_Day7_10_S382_sorted.bam</td>\n",
       "      <td>23B_Day7_10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>23B_Day7_11_S383_sorted.bam</td>\n",
       "      <td>23B_Day7_11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>23B_Day7_12_S384_sorted.bam</td>\n",
       "      <td>23B_Day7_12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>23B_Day7_13_S1294_sorted.bam</td>\n",
       "      <td>23B_Day7_13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>23B_Day7_14_S1295_sorted.bam</td>\n",
       "      <td>23B_Day7_14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1552</th>\n",
       "      <td>30N_Day7_5_S923_sorted.bam</td>\n",
       "      <td>30N_Day7_5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1553</th>\n",
       "      <td>30N_Day7_6_S924_sorted.bam</td>\n",
       "      <td>30N_Day7_6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1554</th>\n",
       "      <td>30N_Day7_7_S925_sorted.bam</td>\n",
       "      <td>30N_Day7_7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1555</th>\n",
       "      <td>30N_Day7_8_S926_sorted.bam</td>\n",
       "      <td>30N_Day7_8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1556</th>\n",
       "      <td>30N_Day7_9_S927_sorted.bam</td>\n",
       "      <td>30N_Day7_9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1557 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                           orig_id    sample-id\n",
       "0      23B_Day7_10_S382_sorted.bam  23B_Day7_10\n",
       "1      23B_Day7_11_S383_sorted.bam  23B_Day7_11\n",
       "2      23B_Day7_12_S384_sorted.bam  23B_Day7_12\n",
       "3     23B_Day7_13_S1294_sorted.bam  23B_Day7_13\n",
       "4     23B_Day7_14_S1295_sorted.bam  23B_Day7_14\n",
       "...                            ...          ...\n",
       "1552    30N_Day7_5_S923_sorted.bam   30N_Day7_5\n",
       "1553    30N_Day7_6_S924_sorted.bam   30N_Day7_6\n",
       "1554    30N_Day7_7_S925_sorted.bam   30N_Day7_7\n",
       "1555    30N_Day7_8_S926_sorted.bam   30N_Day7_8\n",
       "1556    30N_Day7_9_S927_sorted.bam   30N_Day7_9\n",
       "\n",
       "[1557 rows x 2 columns]"
      ]
     },
     "execution_count": 220,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Nissle_sortedbam['sample-id'] = Nissle_sortedbam['orig_id'].str.rsplit(\"_\", n=2).str[0]\n",
    "Nissle_sortedbam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 221,
   "id": "8fddd19f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1536, 2)"
      ]
     },
     "execution_count": 221,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Nissle_sortedbam_final = Nissle_sortedbam[~(Nissle_sortedbam['sample-id'].isin(Nissle_bad))].copy()\n",
    "Nissle_sortedbam_final.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 223,
   "id": "bf0f3004",
   "metadata": {},
   "outputs": [],
   "source": [
    "Nissle_sortedbam_final[['orig_id']].to_csv('AZ20_sortedbam_list_goodones.txt', sep='\\t', index=False, header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 224,
   "id": "7c1eda7f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>orig_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>41B_Day7_10_S73_L002_sorted.bam</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>41B_Day7_11_S81_L002_sorted.bam</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>41B_Day7_12_S89_L002_sorted.bam</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>41B_Day7_13_S2_L002_sorted.bam</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>41B_Day7_14_S10_L002_sorted.bam</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1963</th>\n",
       "      <td>46R_Month6_5_S1093_L003_sorted.bam</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1964</th>\n",
       "      <td>46R_Month6_6_S1101_L003_sorted.bam</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1965</th>\n",
       "      <td>46R_Month6_7_S1109_L003_sorted.bam</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1966</th>\n",
       "      <td>46R_Month6_8_S1117_L003_sorted.bam</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1967</th>\n",
       "      <td>46R_Month6_9_S1125_L003_sorted.bam</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1968 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 orig_id\n",
       "0        41B_Day7_10_S73_L002_sorted.bam\n",
       "1        41B_Day7_11_S81_L002_sorted.bam\n",
       "2        41B_Day7_12_S89_L002_sorted.bam\n",
       "3         41B_Day7_13_S2_L002_sorted.bam\n",
       "4        41B_Day7_14_S10_L002_sorted.bam\n",
       "...                                  ...\n",
       "1963  46R_Month6_5_S1093_L003_sorted.bam\n",
       "1964  46R_Month6_6_S1101_L003_sorted.bam\n",
       "1965  46R_Month6_7_S1109_L003_sorted.bam\n",
       "1966  46R_Month6_8_S1117_L003_sorted.bam\n",
       "1967  46R_Month6_9_S1125_L003_sorted.bam\n",
       "\n",
       "[1968 rows x 1 columns]"
      ]
     },
     "execution_count": 224,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Native_sortedbam = pd.read_csv('AZ51_sortedbam_list_full.txt', sep='\\t', header=None)\n",
    "Native_sortedbam = pd.DataFrame(Native_sortedbam)\n",
    "Native_sortedbam = Native_sortedbam.rename(columns={0:'orig_id'})\n",
    "Native_sortedbam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 226,
   "id": "c03ea18e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>orig_id</th>\n",
       "      <th>sample-id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>41B_Day7_10_S73_L002_sorted.bam</td>\n",
       "      <td>41B_Day7_10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>41B_Day7_11_S81_L002_sorted.bam</td>\n",
       "      <td>41B_Day7_11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>41B_Day7_12_S89_L002_sorted.bam</td>\n",
       "      <td>41B_Day7_12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>41B_Day7_13_S2_L002_sorted.bam</td>\n",
       "      <td>41B_Day7_13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>41B_Day7_14_S10_L002_sorted.bam</td>\n",
       "      <td>41B_Day7_14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1963</th>\n",
       "      <td>46R_Month6_5_S1093_L003_sorted.bam</td>\n",
       "      <td>46R_Month6_5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1964</th>\n",
       "      <td>46R_Month6_6_S1101_L003_sorted.bam</td>\n",
       "      <td>46R_Month6_6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1965</th>\n",
       "      <td>46R_Month6_7_S1109_L003_sorted.bam</td>\n",
       "      <td>46R_Month6_7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1966</th>\n",
       "      <td>46R_Month6_8_S1117_L003_sorted.bam</td>\n",
       "      <td>46R_Month6_8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1967</th>\n",
       "      <td>46R_Month6_9_S1125_L003_sorted.bam</td>\n",
       "      <td>46R_Month6_9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1968 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 orig_id     sample-id\n",
       "0        41B_Day7_10_S73_L002_sorted.bam   41B_Day7_10\n",
       "1        41B_Day7_11_S81_L002_sorted.bam   41B_Day7_11\n",
       "2        41B_Day7_12_S89_L002_sorted.bam   41B_Day7_12\n",
       "3         41B_Day7_13_S2_L002_sorted.bam   41B_Day7_13\n",
       "4        41B_Day7_14_S10_L002_sorted.bam   41B_Day7_14\n",
       "...                                  ...           ...\n",
       "1963  46R_Month6_5_S1093_L003_sorted.bam  46R_Month6_5\n",
       "1964  46R_Month6_6_S1101_L003_sorted.bam  46R_Month6_6\n",
       "1965  46R_Month6_7_S1109_L003_sorted.bam  46R_Month6_7\n",
       "1966  46R_Month6_8_S1117_L003_sorted.bam  46R_Month6_8\n",
       "1967  46R_Month6_9_S1125_L003_sorted.bam  46R_Month6_9\n",
       "\n",
       "[1968 rows x 2 columns]"
      ]
     },
     "execution_count": 226,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Native_sortedbam['sample-id'] = Native_sortedbam['orig_id'].str.rsplit(\"_\", n=3).str[0]\n",
    "Native_sortedbam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 227,
   "id": "2484ef9f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1913, 2)"
      ]
     },
     "execution_count": 227,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Native_sortedbam_final = Native_sortedbam[~(Native_sortedbam['sample-id'].isin(Native_bad))].copy()\n",
    "Native_sortedbam_final.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 228,
   "id": "87ca6652",
   "metadata": {},
   "outputs": [],
   "source": [
    "Native_sortedbam_final[['orig_id']].to_csv('AZ51_sortedbam_list_goodones.txt', sep='\\t', index=False, header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e415996f",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
